# Regression Tasks
## Introduction
This jupyter notebook is made for comparing two models of the regression tasks
which solves the prediction problems which is given below.
## Goals
Our Goal is to predict the last 14 days worth fo sales with the give
data of 2 years. We have two models presented there with the basic
parameters and we have to predict the values and determine which 
model is better for the given data. We have **ARIMA** model and 
**Prophet** model


<br><br>
Now lets see the result of ARIMA model and the value of RMSE (ROOT MEAN SQUARED ERROR).

In [1]:
from models.regression.arima_model.arima_model import arima_model

arima_rmse = arima_model('../../database/database.ini')

Using TensorFlow backend.


Day 1: 23246.663545
Day 2: 19953.698977
Day 3: 12549.208370
Day 4: 15542.696158
Day 5: 15611.828435
Day 6: 17472.059424
Day 7: 31029.318654
Day 8: 13926.235955
Day 9: 23064.097261
Day 10: 17177.871618
Day 11: 14475.652550
Day 12: 17081.563978
Day 13: 26280.196401
Day 14: 37464.321190
RMSE: 32672.368


As you can see the RMSE of arima model is **32672.368**. Now lets check
the RMSE of the prophet model

In [2]:
from models.regression.prophet_model.prophet_model import prophet_model

prophet_rmse = prophet_model('../../database/database.ini')

INFO:fbprophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:fbprophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


         Date          Sale
0  2011-11-24  27763.298484
1  2011-11-25  20600.431665
2  2011-11-27  16328.580585
3  2011-11-28  25268.828016
4  2011-11-29  25112.407953
5  2011-11-30  24357.614432
6  2011-12-01  28098.111742
7  2011-12-02  20935.244923
8  2011-12-04  16663.393843
9  2011-12-05  25603.641274
10 2011-12-06  25447.221211
11 2011-12-07  24692.427690
12 2011-12-08  28432.925000
13 2011-12-09  21270.058181
RMSE: 19653.776


The RMSE of the prophet model is **19653.776** which is far better result
than the ARIMA model. In the both models, there was used basic parameters which
you can change in the defined functions in this project. Evey parameter can be changed
which may produce better results than this one. 